ophyd/pseudopos.py

```diff
@@ -105,20 +149,184 @@ def _started_moving(self, value):
+
+        See `PseudoPositioner.move_single` for more information.
+
+        Parameters
+        ----------
+        pos : float
+            Position to move to
+        kwargs : dict
+            Passed onto parent.move_single()
+        '''
+        self._target = pos
         return self._parent.move_single(self, pos, **kwargs)
 
+    def read(self):
+        d = OrderedDict()
+        d[self.name] = {'value': self.position,
+                        'timestamp': time.time()}
+        return d
+
+    def describe(self):
+        desc = OrderedDict()
+        desc[self.name] = {'dtype': 'number',
+                           'shape': [],
+                           'source': 'computed',
+                           }
+        return desc
+
+    def read_configuration(self):
+        return OrderedDict()
+
+    def describe_configuration(self):
+        return OrderedDict()
 
-class PseudoPositioner(Device, Positioner):
+
+def position_argument_wrapper(type_):
+    '''Wrapper to convert positional arguments to a PositionTuple'''
+    if type_ not in ('pseudo', 'real'):
+        raise ValueError("position_type should be either 'pseudo' or 'real'")
+
+    def wrapper(method):
+        @functools.wraps(method)
+        def wrapped(self, *args, **kwargs):
+            if type_ == 'pseudo':
+                pos, new_kwargs = self.to_pseudo_tuple(*args, **kwargs)
+            else:
+                pos, new_kwargs = self.to_real_tuple(*args, **kwargs)
+
+            return method(self, pos, **new_kwargs)
+
+        return wrapped
+
+    return wrapper
+
+
+real_position_argument = position_argument_wrapper('real')
+pseudo_position_argument = position_argument_wrapper('pseudo')
+
+
+def to_position_tuple(cls, *args, **kwargs):
+    '''Convert user-specified arguments to a Position namedtuple and kwargs
+
+    Example:
+        Tuple = namedtuple('Tuple', 'px py pz')
+
+    All of the following will return the same thing:
+        t, kwargs = to_position_tuple(Tuple, px, py, pz, a=4)
+        t, kwargs = to_position_tuple(Tuple, (px, py, pz), a=4)
+        t, kwargs = to_position_tuple(Tuple, Tuple(px, py, pz), a=4)
+        t, kwargs = to_position_tuple(Tuple, px=1, py=2, pz=3, a=4)
+
+    t will be Tuple(px, py, pz), and kwargs will be {'a': 4}.
+
+    Parameters
+    ----------
+    cls : namedtuple
+        The position class to use. This is likely a RealPosition or a
+        PseudoPosition from a PseudoPositioner.
+    args :
+        User-specified positional arguments
+    kwargs : dict
+        User-specified keyword arguments
+
+    Returns
+    -------
+    position_tuple : cls
+        The position tuple
+    kwargs : dict
+        Keyword arguments not related to the position tuple
+
+    Raises
+    ------
+    TypeError
+        On an empty or invalid namedtuple
+    ValueError
+        On a mismatch of parameters
+    '''
+    usage_info = '''Positions can be passed in a number of ways:
```

Comment:
mild preference to `textwrap.dedent` because I use the left whitespace to see structure in python files.  The other option is to move this block of text to the module level.